### PR TITLE
Enable doctests

### PR DESCRIPTION
**Pull request by RohitRathore1**

### CODE REVIEW

File: src/dataframe/dataframe.jl [hunk 1]
```diff
@@ -91,7 +91,9 @@ functions, or provide type assertions to the variables that hold columns
 extracted from a `DataFrame`.
 
 # Examples
-```julia
+```jldoctest
+julia> using DataFrames
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:95]: I think you should rather use https://juliadocs.github.io/Documenter.jl/stable/man/doctests/#Module-level-metadata for this.

File: src/dataframe/dataframe.jl [hunk 2]
```diff
@@ -1199,17 +1207,14 @@ julia> df2 = DataFrame(A=4.0:6.0, B=4:6)
...
- Row │ A      B
-     │ Int64  Int64
-─────┼──────────────
-   1 │     1      1
-   2 │     2      2
-   3 │     3      3
-   4 │     4      4
-   5 │     5      5
-   6 │     6      6
+julia> df2  # we won't get same result of df1 after appending because append! modifies it's arguments
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:1210]: why do you think it is worth to show `df2` here not `df1` (the point of the example is to show how `df1` was changes after calling `append!`).

Comment (RohitRathore1) [reply src/dataframe/dataframe.jl:1210]: because it was failing in the doctest so I should show `df2` and then in the comment i should tell the reason

Comment (bkamins) [reply src/dataframe/dataframe.jl:1210]: What was the error message?

Comment (RohitRathore1) [reply src/dataframe/dataframe.jl:1210]: The expected output of `df1` should be as we had defined in the starting.

Comment (bkamins) [reply src/dataframe/dataframe.jl:1210]: Please give me the verbatim copy of the error message.

Comment (RohitRathore1) [reply src/dataframe/dataframe.jl:1210]:
```
 ```jldoctest
│ julia> df1 = DataFrame(A=1:3, B=1:3)
│ 3×2 DataFrame
│  Row │ A      B
│      │ Int64  Int64
│ ─────┼──────────────
│    1 │     1      1
│    2 │     2      2
│    3 │     3      3
│
│ julia> df2 = DataFrame(A=4.0:6.0, B=4:6)
│ 3×2 DataFrame
│  Row │ A        B
│      │ Float64  Int64
│ ─────┼────────────────
│    1 │     4.0      4
│    2 │     5.0      5
│    3 │     6.0      6
│
│ julia> append!(df1, df2);
│
│ julia> df1
│ 6×2 DataFrame
│  Row │ A      B
│      │ Int64  Int64
│ ─────┼──────────────
│    1 │     1      1
│    2 │     2      2
│    3 │     3      3
│    4 │     4      4
│    5 │     5      5
│    6 │     6      6
│ ```
│
│ Subexpression:
│
│ df1
│
│ Evaluated output:
│
│ ERROR: UndefVarError: df1 not defined
│
│ Expected output:
│
│ 6×2 DataFrame
│  Row │ A      B
│      │ Int64  Int64
│ ─────┼──────────────
│    1 │     1      1
│    2 │     2      2
│    3 │     3      3
│    4 │     4      4
│    5 │     5      5
│    6 │     6      6
│
│   diff =
│    6×2 DataFrame
│     Row │ A      B
│         │ Int64  Int64
│    ─────┼──────────────
│       1 │     1      1
│       2 │     2      2
│       3 │     3      3
│       4 │     4      4
│       5 │     5      5
│       6 │     6      6ERROR: UndefVarError: df1 not defined
└ @ Documenter.DocTests ~/.julia/packages/Documenter/bFHi4/src/DocTests.jl:373
```

Comment (bkamins) [reply src/dataframe/dataframe.jl:1210]: is this run when we are sure that `using DataFrames` was done before?

@mortenpi - apart from DataFrames.jl not being loaded can you see any other reason why this doctests might fail (it seems to say that `df1` variable is undefined, but it is defined and all works when I copy-paste the example to the terminal).

Comment (mortenpi) [reply src/dataframe/dataframe.jl:1210]: My hunch is that already the `df1 = DataFrame(A=1:3, B=1:3)` block failed to evaluate (e.g. because there was no `using DataFrames`), so `df1` was never set.

Comment (bkamins) [reply src/dataframe/dataframe.jl:1210]: Thank you for looking into this. This was also my guess.

**Comment by RohitRathore1** [comment]
Currently, it resolved only `module not defined or some variable not defined error` so, it's not ready to merge because next step is to resolve the errors in `.md` files

File: src/dataframe/dataframe.jl [hunk 3]
```diff
@@ -1179,6 +1185,8 @@ to vertically concatenate data frames.
 
 # Examples
 ```jldoctest
+julia> using DataFrames
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:1188]: Why is this line needed if you added https://github.com/JuliaData/DataFrames.jl/pull/2703/files#diff-4aae2d1c783cade58bd2cb13748da956e568b7f2aed5fafd9e2a46fb97daf613R2?

Comment (RohitRathore1) [reply src/dataframe/dataframe.jl:1188]: Yeah, I have removed that line in my latest commit.

File: src/abstractdataframe/io.jl [hunk 4]
```diff
@@ -19,15 +19,15 @@ Additionally selected MIME types support passing the following keyword arguments
...
-        \\hline
-        & Int64 & String\\\\
-        \\hline
-        1 & 1 & x \\\\
-        2 & 2 & y \\\\
-        3 & 3 & z \\\\
-\\end{tabular}
+\begin{tabular}{r|cc}
+        & A & B\\
+        \hline
```
Comment (bkamins) [inline src/abstractdataframe/io.jl:24]: this is malformatted

File: src/abstractdataframe/io.jl [hunk 5]
```diff
@@ -19,15 +19,15 @@ Additionally selected MIME types support passing the following keyword arguments
...
+        & Int64 & String\\
+        \hline
```
Comment (bkamins) [inline src/abstractdataframe/io.jl:26]: same here

File: src/abstractdataframe/io.jl [hunk 6]
```diff
@@ -19,15 +19,15 @@ Additionally selected MIME types support passing the following keyword arguments
...
-        & Int64 & String\\\\
-        \\hline
-        1 & 1 & x \\\\
-        2 & 2 & y \\\\
-        3 & 3 & z \\\\
-\\end{tabular}
+\begin{tabular}{r|cc}
+        & A & B\\
+        \hline
+        & Int64 & String\\
```
Comment (bkamins) [inline src/abstractdataframe/io.jl:25]: also here

File: src/abstractdataframe/io.jl [hunk 7]
```diff
@@ -19,15 +19,15 @@ Additionally selected MIME types support passing the following keyword arguments
...
+        \hline
+        1 & 1 & x \\
+        2 & 2 & y \\
+        3 & 3 & z \\
```
Comment (bkamins) [inline src/abstractdataframe/io.jl:29]: and these lines also

File: src/abstractdataframe/io.jl [hunk 8]
```diff
@@ -19,15 +19,15 @@ Additionally selected MIME types support passing the following keyword arguments
...
-        & A & B\\\\
-        \\hline
-        & Int64 & String\\\\
-        \\hline
-        1 & 1 & x \\\\
-        2 & 2 & y \\\\
-        3 & 3 & z \\\\
-\\end{tabular}
+\begin{tabular}{r|cc}
+        & A & B\\
```
Comment (bkamins) [inline src/abstractdataframe/io.jl:23]: this is not OK

File: src/abstractdataframe/io.jl [hunk 9]
```diff
@@ -19,15 +19,15 @@ Additionally selected MIME types support passing the following keyword arguments
...
+        \hline
+        & Int64 & String\\
+        \hline
+        1 & 1 & x \\
+        2 & 2 & y \\
+        3 & 3 & z \\
+\end{tabular}
```
Comment (bkamins) [inline src/abstractdataframe/io.jl:30]: and here

File: src/abstractdataframe/io.jl (hunk 4 above)
Comment (RohitRathore1) [reply src/abstractdataframe/io.jl:24]: how can be solved it?

Comment (bkamins) [reply src/abstractdataframe/io.jl:24]: as commented on slack escape `\`, e.g. `\\begin`

File: docs/make.jl [hunk 10]
```diff
@@ -1,4 +1,5 @@
 using Documenter, DataFrames
+DocMeta.setdocmeta!(DataFrames, :DocTestSetup, :(using DataFrames, CategoricalArrays); recursive=true)
```
Comment (bkamins) [inline docs/make.jl:2]: you must add:
```
using Documenter
using DataFrames
using CategoricalArrays
```
and re-enable the CategoricalArrays.jl tests.

File: docs/Project.toml [hunk 11]
```diff
@@ -1,5 +1,12 @@
 [deps]
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+DataFramesMeta = "1313f7d8-7da2-5740-9ea0-a2ca25f37964"
+CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
+CategoricalArrays = "324d7699-5711-5eae-9e2f-1d82baa6b597"
+Missings = "e1d29d7a-bbdc-5cf2-9ac0-f12de2c33e28"
+Query = "1a8c2f83-1ff3-5112-b086-8aa67b057ba1"
+Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
+
```
Comment (bkamins) [inline docs/Project.toml:9]:
```suggestion
```

File: docs/src/man/missing.md [hunk 12]
```diff
@@ -66,6 +66,8 @@ julia> coalesce.(x, 0)
 The functions `dropmissing` and `dropmissing!` can be used to remove the rows containing `missing` values from a `DataFrame` and either create a new `DataFrame` or mutate the original in-place respectively.
 
 ```jldoctest missings
+julia> using DataFrames
+
```
Comment (bkamins) [inline docs/src/man/missing.md:70]:
```suggestion
```

File: docs/src/man/missing.md [hunk 13]
```diff
@@ -91,6 +93,8 @@ julia> dropmissing(df)
 One can specify the column(s) in which to search for rows containing `missing` values to be removed.
 
 ```jldoctest missings
+julia> using DataFrames
+
```
Comment (bkamins) [inline docs/src/man/missing.md:97]:
```suggestion
```

File: docs/src/man/missing.md [hunk 14]
```diff
@@ -107,6 +111,8 @@ the `Missing` part, if present, set the `disallowmissing` option to `true` (it
 will become the default behavior in the future).
 
 ```jldoctest missings
+julia> using DataFrames
+
```
Comment (bkamins) [inline docs/src/man/missing.md:115]:
```suggestion
```

File: docs/src/man/reshaping_and_pivoting.md [hunk 15]
```diff
@@ -5,6 +5,7 @@ Reshape data from wide to long format using the `stack` function:
 ```jldoctest reshape
 julia> using DataFrames, CSV
 
+julia> iris = DataFrame(CSV.File(joinpath(dirname(pathof(DataFrames)), "../docs/src/assets/iris.csv")))
```
Comment (bkamins) [inline docs/src/man/reshaping_and_pivoting.md:8]:
```suggestion
julia> iris = CSV.read(joinpath(dirname(pathof(DataFrames)), "..", "docs", "src", "assets", "iris.csv")), DataFrame)
```

File: docs/src/man/working_with_dataframes.md [hunk 16]
```diff
@@ -65,7 +68,7 @@ Also notice that when `DataFrame` is printed to the console or rendered in HTML
 (e.g. in Jupyter Notebook) you get an information about type of elements held in
 its columns. For example in this case:
 
-```jldoctest dataframe
+```
```
Comment (bkamins) [inline docs/src/man/working_with_dataframes.md:71]:
```suggestion
```jldoctest dataframe
```

File: docs/src/man/working_with_dataframes.md [hunk 17]
```diff
@@ -784,6 +801,8 @@ column did not allow for missing values.
 
 Replacement operations affecting a single column can be performed using `replace!`:
 ```jldoctest replace
+julia> using DataFrames
+
```
Comment (bkamins) [inline docs/src/man/working_with_dataframes.md:805]:
```suggestion
```

File: src/abstractdataframe/io.jl [hunk 18]
```diff
@@ -17,7 +17,7 @@ Additionally selected MIME types support passing the following keyword arguments
   allows to choose whether to print a brief string summary of the data frame.
 
 # Examples
-```jldoctest
+```
```
Comment (bkamins) [inline src/abstractdataframe/io.jl:20]:
```suggestion
```jldoctest
```

File: src/abstractdataframe/selection.jl [hunk 19]
```diff
@@ -675,7 +675,7 @@ $TRANSFORMATION_COMMON_RULES
   frame or a `GroupedDataFrame`.
 
 # Examples
-```jldoctest
+```
```
Comment (bkamins) [inline src/abstractdataframe/selection.jl:678]:
```suggestion
```jldoctest
```

File: src/abstractdataframe/selection.jl [hunk 20]
```diff
@@ -952,7 +970,7 @@ $TRANSFORMATION_COMMON_RULES
...
```
Comment (bkamins) [inline src/abstractdataframe/selection.jl:973]:
```suggestion
```jldoctest
```

File: src/abstractdataframe/sort.jl [hunk 21]
```diff
@@ -431,7 +431,7 @@ corresponding column index (see example below).
 See other methods for a description of other keyword arguments.
 
 # Examples
-```jldoctest
+```
```
Comment (bkamins) [inline src/abstractdataframe/sort.jl:434]:
```suggestion
```jldoctest
```

File: src/abstractdataframe/selection.jl [hunk 22]
```diff
@@ -776,7 +776,25 @@ julia> df = DataFrame(a = [1, 1, 1, 2, 2, 1, 1, 2],
    7 │     1      2      7
    8 │     2      1      8
 
-julia> gd = groupby(df, :a);
+julia> gd = groupby(df, :a)
```
Comment (bkamins) [inline src/abstractdataframe/selection.jl:779]: why do you want to show `gd`?

File: src/dataframe/dataframe.jl [hunk 23]
```diff
@@ -91,7 +91,9 @@ functions, or provide type assertions to the variables that hold columns
 extracted from a `DataFrame`.
 
 # Examples
-```julia
+```jldoctest
+julia> using DataFrames
+
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:96]:
```suggestion
```

File: src/dataframe/dataframe.jl [hunk 24]
```diff
@@ -774,6 +776,8 @@ If `val` is an `AbstractRange` then the result of `collect(val)` is inserted.
 
 # Examples
 ```jldoctest
+julia> using DataFrames
+
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:780]:
```suggestion
```

File: src/dataframe/dataframe.jl [hunk 25]
```diff
@@ -943,6 +947,8 @@ a vector of sorted and unique integers, a boolean vector, an integer, or `Not`.
...
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:951]:
```suggestion
```

File: src/dataframe/dataframe.jl [hunk 26]
```diff
@@ -1179,6 +1185,8 @@ to vertically concatenate data frames.
...
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:1189]:
```suggestion
```

File: src/dataframe/dataframe.jl (hunk 2 above)
Comment (bkamins) [inline src/dataframe/dataframe.jl:1210]: here `df1` should be shown.

File: src/dataframe/dataframe.jl [hunk 27]
```diff
@@ -1515,6 +1520,8 @@ that are aliases (equal when compared with `===`).
...
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:1524]:
```suggestion
```

File: src/dataframe/dataframe.jl [hunk 28]
```diff
@@ -1633,6 +1640,8 @@ of rows is repeated. Columns of `df` are freshly allocated.
 
 # Example
 ```jldoctest
+julia> using DataFrames
+
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:1644]:
```suggestion
```

File: src/dataframe/dataframe.jl [hunk 29]
```diff
@@ -1676,6 +1685,8 @@ specified by `count`. Columns of `df` are freshly allocated.
...
```
Comment (bkamins) [inline src/dataframe/dataframe.jl:1689]:
```suggestion
```

File: src/dataframe/sort.jl [hunk 30]
```diff
@@ -18,6 +18,8 @@ See other methods for a description of other keyword arguments.
 
 # Examples
 ```jldoctest
+julia> using DataFrames
+
```
Comment (bkamins) [inline src/dataframe/sort.jl:22]:
```suggestion
```

File: docs/src/man/categorical.md [hunk 31]
```diff
@@ -51,21 +50,27 @@ allows us to determine efficiently the allowed levels of the variable at any tim
 the `levels` function (note that levels may or may not be actually used in the data):
 
 ```jldoctest categorical
-julia> levels(cv)
-2-element Array{String,1}:
+julia> levels(cv)
```
Comment (bkamins) [inline docs/src/man/categorical.md:53]:
```suggestion
julia> levels(cv)
```

File: docs/src/man/reshaping_and_pivoting.md [hunk 32]
```diff
@@ -137,49 +186,73 @@ julia> unstack(longdf, :id, :variable, :value)
...
  149 │   149          6.2         3.4          5.4         2.3
  150 │   150          5.9         3.0          5.1         1.8
-                                               143 rows omitted
+                                               135 rows omitted
 ```
 
 If the remaining columns are unique, you can skip the id variable and use:
 
-```jldoctest reshape
+```
```
Comment (bkamins) [inline docs/src/man/reshaping_and_pivoting.md:206]:
```suggestion
```jldoctest reshape
```

File: docs/src/man/reshaping_and_pivoting.md [hunk 33]
```diff
@@ -137,49 +186,73 @@ julia> unstack(longdf, :id, :variable, :value)
...
  149 │ Iris-virginica    149          6.2         3.4          5.4         2.3
  150 │ Iris-virginica    150          5.9         3.0          5.1         1.8
-                                                               143 rows omitted
+                                                                135 rows omitted
 ```
 
 You can even skip passing the `:variable` and `:value` values as positional
 arguments, as they will be used by default, and write:
-```jldoctest reshape
+```
```
Comment (bkamins) [inline docs/src/man/reshaping_and_pivoting.md:233]:
```suggestion
```jldoctest reshape
```

File: docs/src/man/reshaping_and_pivoting.md (hunk 32 above)
Comment (bkamins) [inline docs/src/man/reshaping_and_pivoting.md]:
```suggestion
```jldoctest reshape
```

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Comment by bkamins** [comment]
@nalimilan - we now have doctests enabled in documenter

**Comment by nalimilan** [comment]
Great!
